packages/game-passes/test/GamePasses.test.ts

```diff
@@ -0,0 +1,1251 @@
+        MINT_AMOUNT - 2,
+      );
+      expect(await sandboxPasses.balanceOf(user2.address, TOKEN_ID_2)).to.equal(
+        3,
+      );
+    });
+
+    it('should allow operator to batch burn and mint', async function () {
+      // Mint additional tokens for batch burn
+      await sandboxPasses
+        .connect(admin)
+        .adminMint(user1.address, TOKEN_ID_2, MINT_AMOUNT);
+
+      await sandboxPasses
+        .connect(operator)
+        .operatorBatchBurnAndMint(
+          user1.address,
+          user2.address,
+          [TOKEN_ID_1, TOKEN_ID_2],
+          [2, 3],
+          [TOKEN_ID_2, TOKEN_ID_1],
+          [4, 5],
+        );
+
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_1)).to.equal(
+        MINT_AMOUNT - 2,
+      );
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_2)).to.equal(
+        MINT_AMOUNT - 3,
+      );
+      expect(await sandboxPasses.balanceOf(user2.address, TOKEN_ID_1)).to.equal(
+        5,
+      );
+      expect(await sandboxPasses.balanceOf(user2.address, TOKEN_ID_2)).to.equal(
+        4,
+      );
+    });
+
+    it('should allow user to burn and mint with valid signature', async function () {
+      const deadline = (await time.latest()) + 3600;
+      const nonce = 0;
+
+      // Create signature
+      const signature = await createBurnAndMintSignature(
+        signer,
+        user1.address,
+        TOKEN_ID_1,
+        2,
+        TOKEN_ID_2,
+        3,
+        deadline,
+        nonce,
+      );
+
+      // Burn and mint with signature
+      await sandboxPasses
+        .connect(user1)
+        .burnAndMint(
+          user1.address,
+          TOKEN_ID_1,
+          2,
+          TOKEN_ID_2,
+          3,
+          deadline,
+          signature,
+        );
+
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_1)).to.equal(
+        MINT_AMOUNT - 2,
+      );
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_2)).to.equal(
+        3,
+      );
+    });
+  });
+
+  describe('Transfer Restrictions', function () {
+    beforeEach(async function () {
+      // Mint transferable and non-transferable tokens
+      await sandboxPasses
+        .connect(admin)
+        .adminMint(user1.address, TOKEN_ID_1, MINT_AMOUNT); // Transferable
+      await sandboxPasses
+        .connect(admin)
+        .adminMint(user1.address, TOKEN_ID_2, MINT_AMOUNT); // Non-transferable
+    });
+
+    it('should allow transferring transferable tokens', async function () {
+      await sandboxPasses
+        .connect(user1)
+        .safeTransferFrom(user1.address, user2.address, TOKEN_ID_1, 2, '0x');
+
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_1)).to.equal(
+        MINT_AMOUNT - 2,
+      );
+      expect(await sandboxPasses.balanceOf(user2.address, TOKEN_ID_1)).to.equal(
+        2,
+      );
+    });
+
+    it('should not allow transferring non-transferable tokens', async function () {
```

Comment:
presumably this should also apply to burn and mint operations - let's add tests for these scenarios

Reply:
test name should read `if not admin`

Reply:
No burning and minting does not relate to transferability.
address(0) => user and user => address(0) is always allowed

Updated text name https://github.com/thesandboxgame/sandbox-smart-contracts/pull/1619/commits/4d36bc4eaf584e0dcb728aa9eee40a5574fabc4c